Counterexamples/CliffordAlgebra_not_injective.lean

```diff
@@ -0,0 +1,317 @@
+      Finsupp.add_apply, Finsupp.equivFunOnFinite_symm_apply_toFun] at h
+    refine' (h _ hm).imp fun i hi => ⟨Set.mem_univ _, _⟩
+    rintro hmi
+    rw [hmi] at hi
+    norm_num at hi
+  rw [as_sum x, CharTwo.sum_mul_self]
+  refine' sum_mem fun m hm => _
+  rw [mem_kIdeal_iff, monomial_mul]
+  intro m' hm'
+  obtain rfl := Finset.mem_singleton.1 (support_monomial_subset hm')
+  rw [mem_ideal_span_X_image] at this
+  obtain ⟨i, _, hi⟩ := this m hm
+  simp_rw [←one_add_one_eq_two]
+  refine' ⟨i, Nat.add_le_add _ _⟩ <;> rwa [Nat.one_le_iff_ne_zero]
+
+/-- `𝔽₂[α, β, γ] / (α², β², γ²)` -/
+def K : Type _ := _ ⧸ kIdeal
+
+instance : CommRing K := by dsimp only [K]; infer_instance
+instance : CommSemiring K := by dsimp only [K]; infer_instance
+instance : Ring K := by dsimp only [K]; infer_instance
+instance : Semiring K := by dsimp only [K]; infer_instance
+instance : AddCommGroup K := by dsimp only [K]; infer_instance
+instance : AddCommMonoid K := by dsimp only [K]; infer_instance
+
+theorem comap_C_span_le_bot : kIdeal.comap (C : ZMod 2 →+* MvPolynomial (Fin 3) (ZMod 2)) ≤ ⊥ := by
+  refine' (Ideal.comap_span_le _ _ (constantCoeff_C _) _).trans _
+  refine' (Ideal.span_le_bot _).2 _
+  rintro x ⟨_, ⟨i, rfl⟩, rfl⟩
+  rw [RingHom.map_mul, constantCoeff_X, MulZeroClass.mul_zero, Set.mem_singleton_iff]
+
+/-- `k` has characteristic 2. -/
+instance K.charP : CharP K 2 := by
+  dsimp only [K]
+  rw [CharP.quotient_iff'']
+  have : Nat.castRingHom (MvPolynomial (Fin 3) (ZMod 2)) = C.comp (Nat.castRingHom _) := by
+    ext1 r; rfl
+  rw [this, ← Ideal.comap_comap, ← RingHom.comap_ker]
+  exact Ideal.comap_mono (comap_C_span_le_bot.trans bot_le)
+
+/-- The generators of `K`. -/
+def K.gen (i : Fin 3) : K := Ideal.Quotient.mk _ (MvPolynomial.X i)
+
+local notation "α" => K.gen 0
+local notation "β" => K.gen 1
+local notation "γ" => K.gen 2
+
+/-- The elements above square to zero -/
+@[simp]
+theorem X_sq (i : Fin 3) : K.gen i * K.gen i = (0 : K) := by
+  change Ideal.Quotient.mk _ _ = _
+  rw [Ideal.Quotient.eq_zero_iff_mem]
+  exact Ideal.subset_span ⟨i, rfl⟩
+
+/-- If an element multiplied by `αβγ` is zero then it squares to zero. -/
+theorem sq_zero_of_αβγ_mul {x : K} : α * β * γ * x = 0 → x * x = 0 := by
+  induction x using Quotient.inductionOn'
+  change Ideal.Quotient.mk _ _ = 0 → Ideal.Quotient.mk _ _ = 0
+  rw [Ideal.Quotient.eq_zero_iff_mem, Ideal.Quotient.eq_zero_iff_mem]
+  exact mul_self_mem_kIdeal_of_X0_X1_X2_mul_mem
+
+/-- Though `αβγ` is not itself zero-/
+theorem αβγ_ne_zero : α * β * γ ≠ 0 := fun h =>
+  X0_X1_X2_nmem_kIdeal <| Ideal.Quotient.eq_zero_iff_mem.1 h
+
+-- A variant of lean4#2220
+local macro_rules | `($x • $y) => `(@HSMul.hSMul _ _ _ instHSMul $x $y)
+
+/-- The 1-form on $K^3$, the kernel of which we will take a quotient by. -/
+@[simps!]
+def lFunc : (Fin 3 → K) →ₗ[K] K :=
+  letI proj : Fin 3 → (Fin 3 → K) →ₗ[K] K := LinearMap.proj
+  α • proj 0 - β • proj 1 - γ • proj 2
```

Comment:
Why subtract when you can add? We're in characteristic 2. Changing `-` to `+` doesn't break anything.

Reply:
I only did this because it matches the source material. Indeed, it makes little difference. Do you want me to make the change (and add a comment explaining why we changed from Darij's informal version)?

Reply:
I made the change; when you said "doesn't break anything", I thought you meant "requires minimal changes" not "goes through for free"!